lib/binding_rust/lib.rs

```diff
@@ -752,7 +767,7 @@ impl Parser {
         old_tree: Option<&Tree>,
         options: Option<ParseOptions>,
     ) -> Option<Tree> {
-        type Payload<'a, F, T> = (&'a mut F, Option<T>);
+        type Payload<'a, F, T> = (&'a mut F, Option<DecodeCallback<'a>>, Option<T>);
```

Comment:
Overall, I don't get why the decode callback would be part of the Options. It doesn't really make sense to say `parse_utf16` with a custom payload. Same thing for the other `parse` APIs, which are otherwise documented as taking UTF8 input.